src/interactions/slashCommands/mixins/CommandChannelOptionBase.ts

```diff
@@ -16,7 +16,7 @@ const allowedChannelTypes = [
 	ChannelType.GuildPrivateThread,
 ];
 
-const channelTypePredicate = ow.number.oneOf(allowedChannelTypes);
+const channelTypePredicate = z.union(allowedChannelTypes.map((type) => z.literal(type)) as any);
```

Comment:
What's the justification for this cast?

Reply:
Even though `allowedChannelTypes.map((type) => z.literal(type))` returns what z.union wants, an array of zod types with at least 2 elements, TS complains with `Source provides no match for required element at position 0 in target.` It seems like it can't infer that allowedChannelTypes isn't an empty array so the cast is to tell it it'll work fine. As far as casting to any, I meant to change it to a more specific cast but forgot before opening the pr.